scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -210,6 +211,9 @@ trait TypedPipe[+T] extends Serializable {
   /** prints the current pipe to stdout */
   def debug: TypedPipe[T] = onRawSingle(_.debug)
 
+  /** adds a description to the pipe */
+  def withDescription(description: String): TypedPipe[T] = new WithDescriptionTypedPipe[T](this, description)
```

Comment:
I liked this approach because it just wrapped a pipe with a description, and you can see how easy it was.  otherwise every version of TypedPipe is going to have to set the description in their ".toPipe" ... this puts it all in one place and it didn't seem as we lose any functionality...  it is analogous to the way we are doing WithOnComplete on TypedPipe